test/primitives/base-element-test.js

```diff
@@ -231,4 +231,158 @@ describe('BaseElement', function() {
       });
     });
   });
+
+  context('when querying', function() {
+    it('returns null when there are no matching elements', function() {
+      const element = new minim.BaseElement();
+      const result = element.findRecursive('string');
+
+      expect(result).to.equal(null);
+    });
+
+    it('finds direct element', function() {
+      const StringElement = minim.getElementClass('string');
+      const element = new minim.BaseElement(
+        new StringElement('Hello World')
+      );
+
+      const result = element.findRecursive('string');
+
+      expect(result.element).to.equal('array');
+      expect(result.toValue()).to.deep.equal(['Hello World']);
+    });
+
```

Comment:
what would be a result of `new StringElement('Hello World').findRecursive('string')`?